django/utils/dateparse.py

```diff
@@ -156,3 +156,16 @@ def parse_duration(value):
         if match.re == iso8601_duration_re:
             days *= sign
         return days + sign * datetime.timedelta(**kw)
+
+
+def parse_timezone_delta(tzname):
+    """
+    Parse delta time in timezone name
+
+    Return sign index and delta value.
+    """
+    for sign in ("+", "-"):
+        if sign in tzname and parse_time(tzname.split(sign)[-1]):
+            return tzname.find(sign), tzname[tzname.find(sign):]
```

Comment:
We could return `tzname` and `offset` instead. This should simplify `_sqlite_datetime_parse()` and `_prepare_tzname_delta()`, e.g.
```python
def split_timezone_delta(tzname):
    for sign in ('+', '-'):
        if sign in tzname:
            name, offset = tzname.rsplit(sign, 1)
            if offset and parse_time(offset):
                return name, f'{sign}{offset}'
    return tzname, None
```